### ISSUE TITLE
support for (more) general-site generator

### ISSUE THREAD

**Issue by gour** [greghendershott/frog#123]
Hello,

at the moment I use [Hugo](http://gohugo.io/) static site generator written in Go which is very nice & flexible.

However, I've become interested in Racket, so, naturally, considering whether Frog can replace it.

One nice feature in Hugo is e.g. support for [Taxonomies](http://gohugo.io/taxonomies/overview) or [Shortcodes](http://gohugo.io/extras/shortcodes/), so I wonder if you have plan to extend Frog to be more general site-oriented generator (e.g. nested-level menus, theme support etc.) instead of more blog-oriented one with the capabilities for general posts?

(Maybe Frog already can support those, but I'm too new to discover...)

I'm still waiting for next release of Hugo to support using of Asciidoc(tor) markup considering that it is more suitable than Markdown for larger or more complex posts/docs. Otoh, Frog has support for Scribble which, afaict, is very capable, so taking advantage of it **could** make Frog much more capable and it could also serve to attract new people to Racket. (Hugo is one of the most popular Go projects at Github.)

What do you think?

**Comment by greghendershott** [comment]
Good question. The name Frog comes from <b>fr</b>ozen bl<b>og</b>. The goal is "just works" blogging. Otherwise I'd have named it Frite -- <b>fr</b>ozen s<b>ite</b>. :)

I don't have plans to make a general purpose static site generator. If I did, I'd probably start with a fresh design, as I mention [here](https://github.com/greghendershott/frog/issues/58).

Depending on what you want to do, you might find [Pollen](https://github.com/mbutterick/pollen) to be a better choice. Although it's also not trying to bit a general static site generator, it is a book generator. That might cover more of what you need.

p.s. IIUC "shortcodes" are a subset of Scribble. In Scribble you're not limited to some sort of template language. You have all of Racket available. In fact Scribble documents really are Racket programs.

**Comment by gour** [comment]
> Good question. The name Frog comes from frozen blog. The goal is "just works" blogging.

OK. I got it. ;)

>  Otherwise I'd have named it Frite -- frozen site. :)

As vegetarian, I prefer frite over frogs, but stll fresh instead of frozen. :-)

> Depending on what you want to do, you might find Pollen to be a better choice. Although it's also not trying to be a general static site generator, it is a book generator. That might cover more of what you need.

I've it on my list and the author's article has brought me to the Racket, still I've to see whether it can serve the purpose.
> p.s. IIUC "shortcodes" are a subset of Scribble. In Scribble you're not limited to some sort of template language. You have all of Racket available. In fact Scribble documents really are Racket programs.

Well, that's why I find interesting to use Scribble for writing web content as well.

**Comment by jbshirk** [comment]
@greghendershott wrote:

> In Scribble you're not limited to some sort of template language. You have all of Racket available. In fact Scribble documents really are Racket programs.


https://github.com/mbutterick/pollen/issues/43#issuecomment-410076068

**Comment by kisaragi-hiu** [comment]
@jbshirk You can use Pollen as a preprocessor before running frog or implement your own blogging utilities in Pollen.

Check out [kisaragi-hiu.com (disclaimer: this is my blog)](https://github.com/kisaragi-hiu/kisaragi-hiu.github.io) for the Pollen preprocessor approach, [mstill.io](https://github.com/malcolmstill/mstill.io) for a simpler blog built in Pollen, or [thenotepad.org](https://github.com/otherjoel/thenotepad) for a more full-featured blog utilizing many Pollen.

Maybe searching for `frog pollen Racket`, `pollen racket blog`, `frog, the frozen blog generator` or something else can yield other examples.

**TL;DR: no, not really, there are ways to utilize Pollen while using Frog.**

**Comment by jbshirk** [comment]
thenotepad.org is gold. Thank you!
